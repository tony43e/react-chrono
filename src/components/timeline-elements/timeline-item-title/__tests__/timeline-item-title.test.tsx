import { render } from "@testing-library/react";
import React from "react";
import Title from "../timeline-card-title";

test("Check Timeline Item Title", () => {
  const { getByText } = render(<Title title="Test" />);
  const titleElement = getByText(/test/i);
  expect(titleElement).toBeInTheDocument();
});

test("Check Timeline Item Title Snapshot", () => {
  const { getByText } = render(<Title title="Test" active />);
  const titleElement = getByText(/test/i);
  expect(titleElement).toMatchSnapshot();
});

test("Check Timeline Active", () => {
  const { getByText } = render(<Title title="Test" active />);
  const titleElement = getByText(/test/i);
  expect(titleElement).toHaveClass("active");
});

test("Check Timeline Not Active", () => {
  const { getByText } = render(<Title title="Test" active={false} />);
  const titleElement = getByText(/test/i);
  expect(titleElement).not.toHaveClass("active");
});
