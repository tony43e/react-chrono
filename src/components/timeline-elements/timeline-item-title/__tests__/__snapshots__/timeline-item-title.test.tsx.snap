// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check Timeline Item Title Snapshot 1`] = `
<div
  class="sc-bdnylx dzAmsA active"
  title="Test"
>
  Test
</div>
`;
